File: java/EngineerManager.java
// You need to include the following dependencies: AWS Lambda Java Core, JDBC, HikariCP, and any other required library

import com.amazonaws.services.lambda.runtime.Context;
import com.amazonaws.services.lambda.runtime.RequestHandler;
import com.amazonaws.services.lambda.runtime.LambdaLogger;
import com.zaxxer.hikari.HikariDataSource;
import com.zaxxer.hikari.HikariConfig;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.util.ArrayList;
import java.util.List;

public class EngineerManager implements RequestHandler<RequestClass, ResponseClass> {
  
  // Initialize a connection pool (HikariCP)
  private static final HikariConfig config = new HikariConfig();
  private static final HikariDataSource ds;
  
  static {
    config.setJdbcUrl("jdbc:mysql://your-rds-instance:3306/yourdatabase");
    config.setUsername("yourusername");
    config.setPassword("yourpassword");
    config.addDataSourceProperty("cachePrepStmts", "true");
    config.addDataSourceProperty("prepStmtCacheSize", "250");
    config.addDataSourceProperty("prepStmtCacheSqlLimit", "2048");
    ds = new HikariDataSource(config);
  }

  public ResponseClass handleRequest(RequestClass input, Context context) {
    LambdaLogger logger = context.getLogger();
    logger.log("Input: " + input);

    // Based on the input, determine which operation to perform
    // For example:
    if ("createEngineer".equals(input.getAction())) {
      // ... call createEngineer method
    }
    // Other operations (updateEngineer, getEngineerById, getEngineerAll)
    // ...

    return new ResponseClass();
  }

  // Method to create an engineer
  private void createEngineer(Engineer engineer) {
    // SQL query to insert a new engineer record
    String sql = "INSERT INTO engineers (userId, firstName, lastName, title, statusId, siteId) VALUES (?, ?, ?, ?, ?, ?);";

    // Use the connection pool to get a connection
    try (Connection conn = ds.getConnection();
         PreparedStatement stmt = conn.prepareStatement(sql)) {
      stmt.setInt(1, engineer.getUserId());
      stmt.setString(2, engineer.getFirstName());
      stmt.setString(3, engineer.getLastName());
      stmt.setString(4, engineer.getTitle());
      stmt.setInt(5, engineer.getStatusId());
      stmt.setInt(6, engineer.getSiteId());
      stmt.executeUpdate();
    } catch (Exception e) {
      e.printStackTrace();
      // Handle exceptions
    }
  }

  // Methods for updateEngineer, getEngineerById, getEngineerAll would follow a similar pattern
  
  // ... (additional methods)

  // Engineer interface
  public interface Engineer {
    int getUserId();
    String getFirstName();
    String getLastName();
    String getTitle();
    int getStatusId();
    int getSiteId();
    String getSiteName();
    String getStatus();
  }
  
  // RequestClass and ResponseClass need to be implemented
  // ...
}

// You would also need to create POJOs for RequestClass and ResponseClass to represent incoming requests and outgoing responses

File: nodejs/EngineerManager.js
// First, you need to install the `pg` package using npm or yarn
// npm install pg

// Importing the AWS SDK and pg for PostgreSQL connection
import AWS from 'aws-sdk';
import { Pool } from 'pg';

// Define database connection details
const pool = new Pool({
  user: 'your_database_user',
  host: 'your_database_host',
  database: 'your_database_name',
  password: 'your_database_password',
  port: 'your_database_port',
});

// Lambda function to create an engineer
export const createEngineer = async (event) => {
  const engineer = JSON.parse(event.body); // Assuming the body is JSON
  const query = 'INSERT INTO engineers(userId, firstName, lastName, title, statusId, siteId) VALUES($1, $2, $3, $4, $5, $6)';
  const values = [engineer.userId, engineer.firstName, engineer.lastName, engineer.title, engineer.statusId, engineer.siteId];

  try {
    await pool.query(query, values);
    return { statusCode: 201, body: JSON.stringify({ message: 'Engineer created' }) };
  } catch (err) {
    return { statusCode: 500, body: JSON.stringify({ error: err.message }) };
  }
};

// Other CRUD operations (updateEngineer, getEngineerById, getEngineerAll) will be implemented similarly.

